docs/zkapps/tutorials/oracle.mdx

```diff
@@ -0,0 +1,482 @@
+---
+title: 'Example: Oracle'
+hide_title: true
+sidebar_label: 'Example: Oracle'
+---
+
+:::info
+
+Please note that zkApp programmability is not yet available on Mina Mainnet, but
+zkApps can now be deployed to Berkeley Testnet.
+
+:::
+
+# Example: Oracle
+
+## Overview
+
+If your smart contract needs to consume data from the outside world, you can use an oracle.
+
+In this tutorial, we’ll build an oracle that retrieves data from a REST API. We’ll also write a smart contract that consumes information from this oracle.
+
+Specifically, our oracle will retrieve data from a REST API that provides mock credit score information for two users–one with a high credit score and one with a low credit score. Our smart contract will consume this information and allow the user to prove their credit score is above a certain threshold (e.g. >700).
+
+Using our smart contract, an end user will be able to generate an attestation that their credit score is above a certain value, which they can present to a third party to prove this fact without needing to share the exact credit score or other personal info with this 3rd party, maintaining their privacy.
+
+Although we’re using a mock credit score API as our data source, you can easily swap out the oracle’s HTTP request to use any REST API or GraphQL API source of your choice, to create an oracle for any type of data.
+
+## Oracle
+
+### How it works
+
+Let’s start by understanding how an oracle works.
+
+Mina smart contracts run off-chain and make it possible to prove that the expected computation was run on private data without revealing the data itself. But if the smart contract consumes data from a 3rd party source, we need a way to verify that this data is authentic–i.e. that it was provided by the expected source.
+
+In the future, Mina’s zkOracles will allow a zkApp to consume data trustlessly from any HTTPS data source. But for now, the oracle design described in this tutorial will typically be operated by the zkApp developer, it will fetch and sign the desired data, and then a zkApp can consume this data and will verify the signature to ensure that the data was provided by the expected source.
+
```

Comment:
This also allows for data providers themselves to sign the data, without the zkApp developer having to fetch and act as an intermediary. In this way, when a KYC or credit score provider starts to sign their API calls, zkApp developers can easily consume that data while ensuring it is from the expected source.